[raynertjx] iP

raynertjx opened the pull request:
### BongoBot
> "Bongo Cat is very cute." -raynertjx

BongoBot frees your mind of having to remember things you need to do. It's:
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST TO USE

All you need to do is:
1. download it from [here](https://github.com/raynertjx/ip/releases/tag/v0.1).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😸

And it is completely **FREE**!

Features:
- [x] Managing todos/deadlines/events
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Bongo.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

raynertjx committed:
added rename, greet, exit

raynertjx committed:
added echo

raynertjx committed:
added add, list for tasks

raynertjx committed:
added mark as done

raynertjx committed:
added todos, events, deadlines

raynertjx committed:
added automated text ui testing

raynertjx committed:
added error handling

raynertjx committed:
added delete

raynertjx committed:
changed to switch statement

raynertjx committed:
Rename files from Duke to Bongo

raynertjx committed:
Add initial tasks

raynertjx committed:
Rename duke.txt to bongo.txt

raynertjx committed:
Add additional constructor for respective tasks

raynertjx committed:
Abstract main method into helper methods

raynertjx committed:
Change access modifiers to default

raynertjx committed:
Add saving of tasks when task list changes

raynertjx committed:
Add error handling if directory or file does not exist

raynertjx committed:
Edit files to use new name Bongo

raynertjx committed:
Merge branch 'branch-Level-7'

raynertjx committed:
Add generateDateString helper method

raynertjx committed:
Store date and times in LocalDateTime format

raynertjx committed:
Add error handling for invalid datetimes

raynertjx committed:
Merge branch 'branch-Level-8'

raynertjx committed:
Add Ui class and its methods

raynertjx committed:
Add Ui constructor and readCommand method

raynertjx committed:
Add Storage class and its methods

raynertjx committed:
Add TaskList class and its methods

raynertjx committed:
Add Parser class and its methods

raynertjx committed:
Initialise Command classes

raynertjx committed:
Add implementations for abstract execute method

raynertjx committed:
Add isExit boolean field

raynertjx committed:
Add utility class for date formatting

raynertjx committed:
Edit default error message

raynertjx committed:
Change methods to throw BongoException

raynertjx committed:
Adjust line spacing and add loading error messages

raynertjx committed:
Change access modifiers to protected

raynertjx committed:
Add method to handle case where files not found

raynertjx committed:
Remove error handling and shift it to Command instances

raynertjx committed:
Change to use extracted classes and its abstracted methods

raynertjx committed:
Merge branch 'branch-A-MoreOOP'

raynertjx committed:
Add data directory

raynertjx committed:
Divide classes into packages

raynertjx committed:
Fix typo in generateStringForTextFile method

raynertjx committed:
Edit error message

raynertjx committed:
Edit import statements

raynertjx committed:
Merge branch 'branch-A-Packages'

raynertjx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

raynertjx committed:
Add Gradle build

raynertjx committed:
Merge branch 'branch-A-Gradle'

raynertjx committed:
Shift Bongo class to under bongo package

raynertjx committed:
Merge branch 'branch-A-Gradle'

raynertjx committed:
Initialise BongoTest

raynertjx committed:
Change Parser class to static

raynertjx committed:
Edit error messages

raynertjx committed:
Modify exception handling and error message

raynertjx committed:
Add tests for deadlines

raynertjx committed:
Edit error handling and shift into switch block

raynertjx committed:
Edit deadline unit tests

raynertjx committed:
Add unit tests for events

raynertjx committed:
Add unit tests for todos

raynertjx committed:
Merge branch 'branch-A-JUnit'

raynertjx committed:
Package into jar file

raynertjx committed:
Merge branch 'branch-A-Jar'

raynertjx committed:
Add Javadoc comments for task package

raynertjx committed:
Add Javadoc comments for Bongo class

raynertjx committed:
Add Javadoc comments for helper package

raynertjx committed:
Add Javadoc comments for command package

raynertjx committed:
Change code to follow coding standard

raynertjx committed:
Add find task functionality

raynertjx committed:
Fix bug when searchInput is null

raynertjx committed:
Merge branch 'branch-A-JavaDoc'

raynertjx committed:
Merge branch 'branch-A-CodingStandard'

raynertjx committed:
Add more Javadocs comments

raynertjx committed:
Edit parsing and exception handling of mark, unmark and delete

raynertjx committed:
Merge branch 'branch-Level-9'

raynertjx committed:
Remove merge conflict comments

raynertjx committed:
Add config files

raynertjx committed:
Add checkstyle config to gradle build file

raynertjx committed:
Prevent loading of expired tasks

raynertjx committed:
Resolve style violations generated by checkstyleMain

raynertjx committed:
Delete expired tasks from text file upon loading

raynertjx committed:
Merge branch 'branch-A-CheckStyle'

raynertjx committed:
Set up Java FX

raynertjx committed:
Add images

raynertjx committed:
Finish Iteration 1

raynertjx committed:
Create new images sub directory

raynertjx committed:
Finish Iteration 2

raynertjx committed:
Finish Iteration 3

raynertjx committed:
Add FXML support in application

raynertjx committed:
Change method signature of execute in Command

raynertjx committed:
Add custom GUI

raynertjx committed:
Merge branch 'branch-Level-10'

ricketytoc commented on src/main/java/bongo/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+            if (index == -1) {
+                throw new BongoException("Please include the datetime of your deadline.");
+            }
+            String deadlineDesc = taskInput.substring(0, index - 1);
```
Perhaps you could rename deadlineDesc to deadlineDescription as when I first read it I thought just by reading the variable name it meant deadlineDescending.

ricketytoc commented on src/main/java/bongo/helper/DateHelper.java:
```diff
@@ -0,0 +1,46 @@
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+
+    public static boolean hasTaskExpired(LocalDateTime datetime) {
+        return datetime.isBefore(LocalDateTime.now());
```
I like the naming of the isBefore() method , it makes it very easy to read!

ricketytoc commented on src/main/java/bongo/helper/Storage.java:
```diff
@@ -0,0 +1,223 @@
+    public ArrayList<Task> load() throws FileNotFoundException, BongoException {
+        File file = new File(this.pathname);
+        Scanner fileScanner = new Scanner(file);
+        ArrayList<Task> loadedTasks = new ArrayList<>();
```
Good job naming the variable name for ArrayList with multiple items in plural form!

ricketytoc commented on src/main/java/bongo/helper/Ui.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    /**
+     * Shows all tasks to the user.
```
Good job following the coding standard for java docs description writing!

ricketytoc reviewed (commented):
Great job! Code is very readable to me

Ken-Lai commented on src/main/java/bongo/helper/Storage.java:
```diff
@@ -0,0 +1,223 @@
+                arr[i] = arr[i].trim();
+            }
+            boolean isTaskMarkedDone = arr[1].equals("1");
+            LocalDateTime datetime;
```
Perhaps change to dateTime instead?

Ken-Lai reviewed (commented):
Overall code has no issues, just one minor naming issue. Good job!

raynertjx committed:
Only exceptions are being used to handle errors.

Assertions can help verify certain assumptions in the code, and can
complement exceptions to handle errors in the code.

Inserting assertions into help verify that no invalid inputs are
being passed into Commands and that a response will be shown to
the user.

Assertions and exceptions complement each other to handle errors.

raynertjx committed:
Code styling and quality fixes

raynertjx committed:
Change execute method to use streams

raynertjx committed:
Merge pull request #1 from raynertjx/branch-A-Assertions

Add assertions

raynertjx committed:
Merge branch 'master' into branch-A-CodeQuality

raynertjx committed:
Merge branch 'master' into branch-A-Streams

raynertjx committed:
Merge pull request #2 from raynertjx/branch-A-CodeQuality

Code styling and quality fixes

raynertjx committed:
Merge branch 'master' into branch-A-Streams

raynertjx committed:
Merge pull request #3 from raynertjx/branch-A-Streams

The execute method in the FindCommand class uses a traditional for-loop
to filter tasks based on a search parameter.

Utilising Java streams for filtering tasks provides a more concise and
expressive way to perform the same operation.

I have refactored the execute method to use Java streams, enhancing code
readability and maintainability.

Java streams simplify the code, makign it more efficient and easier to
understand.

raynertjx committed:
Add yml file

raynertjx committed:
Fix code styling

raynertjx committed:
Fix code styling for tests

raynertjx committed:
Merge pull request #4 from raynertjx/branch-A-CI

Add CI

raynertjx committed:
Add isTaskScheduledForDate method

raynertjx committed:
Add ability to view schedules on a specific date

raynertjx committed:
Merge branch 'branch-B-ViewSchedules'

raynertjx committed:
Add bot title

raynertjx committed:
Add user guide for Bongo

raynertjx committed:
Merge branch 'branch-A-UserGuide'
